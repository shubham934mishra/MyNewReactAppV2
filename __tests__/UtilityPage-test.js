import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import UtilityPage from '../src/components/Utility/UtilityScreen';

import {
  mergingTwoObjects,
    mergingTwoArrays,
    concatNumberAndString
} from '../src/utils/Utility';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ LoginReducer: { data: 'shubham' } });

const baseProps = {
    navigation: {
      navigate: jest.fn(),
      store: jest.fn()
    }
  };


  
test('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}>
      <UtilityPage {...baseProps} />
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  // 2 Objects
const obj1 = {test:'testCase1',value:90};
const obj2 = {name:'Shubham',age:25};
const mergedObj = {...obj1,...obj2};

// 2 Arrays
const arr1 = [1,2,3,4,5];
const arr2 = ['arr1','arr2','arr3','arr4','arr5'];
const mergedArr = [...arr1,...arr2];

// 2 Parameters {number | string}
const param1 = 40;
const param2 = 90;

describe('Checking merging of Two Objects', () => {
    test('given two objects, mergingTwoObjects returns a merged object', () => {
    expect(mergingTwoObjects(obj1,obj2)).toEqual(mergedObj);
    });

    test('given one object, mergingTwoObjects received undefined', () => {
    expect(mergingTwoObjects(obj1)).toBeUndefined();
    });

    test('given one object and null, mergingTwoObjects received undefined', () => {
    expect(mergingTwoObjects(obj1,null)).toBeUndefined();
    });

    test('given no parameter, mergingTwoObjects received undefined', () => {
    expect(mergingTwoObjects()).toBeUndefined();
    });
});

describe('Checking mergingTwoArrays ', () => {
    test('given two arrays, mergingTwoArrays returns a merged object', () => {
    expect(mergingTwoArrays(arr1,arr2)).toEqual(mergedArr);
    });

    test('given only one array, mergingTwoArrays received undefined', () => {
    expect(mergingTwoArrays(arr1)).toBeUndefined();
    });

    test('given one array and null, mergingTwoArrays received undefined', () => {
    expect(mergingTwoArrays(arr1,null)).toBeUndefined();
    });

    test('given no parameter, mergingTwoArrays received undefined', () => {
    expect(mergingTwoArrays()).toBeUndefined();
    });
});

describe('Checking concatination of Number nd String', () => {
    test('given two numbers, concatNumberAndString returns an addition value', () => {
    expect(concatNumberAndString(param1,param2)).toEqual(130);
    });

    test('given one number one String, concatNumberAndString received concatenated Value', () => {
    expect(concatNumberAndString(param1,param2.toString())).toMatch(`${param1}90`);
    });

    test('given both are String, concatNumberAndString received concatenated Value', () => {
    expect(concatNumberAndString(param1.toString(),param2.toString())).toMatch(`${param1}${param2}`);
    });

    test('given one number, concatNumberAndString received undefined', () => {
    expect(isNaN(concatNumberAndString(param1)));
    });

    test('given one number and null, concatNumberAndString received undefined', () => {
    expect(isNaN(concatNumberAndString(param1,null)));
    });

    test('given no parameter, concatNumberAndString received undefined', () => {
    expect(isNaN(concatNumberAndString()));
    });
});

