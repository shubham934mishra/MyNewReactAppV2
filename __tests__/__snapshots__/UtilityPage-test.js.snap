// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1a1924",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 22,
        "height": 30,
        "marginBottom": 20,
      }
    }
  >
    Welcome,
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#1a1924",
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 22,
          "height": 30,
          "marginBottom": 20,
        }
      }
    >
      Merging The objects:
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      1st Object : 
      {"test":"testCase1","value":90}
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      2nd Object : 
      {"name":"Shubham","age":25}
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      Results : 
      {"test":"testCase1","value":90,"name":"Shubham","age":25}
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1a1924",
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 22,
          "height": 30,
          "marginBottom": 20,
        }
      }
    >
      Merging The Arrays:
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      1st Array : 
      [1,2,3,4,5]
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      2nd Array : 
      ["arr1","arr2","arr3","arr4","arr5"]
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      Results : 
      [1,2,3,4,5,"arr1","arr2","arr3","arr4","arr5"]
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1a1924",
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 22,
          "height": 30,
          "marginBottom": 20,
        }
      }
    >
      Concatination:
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      1st 
      number
       : 
      40
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      2nd 
      number
       : 
      90
       
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 15,
        }
      }
    >
      Results : 
      130
    </Text>
  </View>
</RCTSafeAreaView>
`;
